internal/bgp/frr/frr.go

```diff
@@ -44,8 +44,11 @@ var osHostname = os.Hostname
 
 // sessionName() defines the format of the key of the 'sessions' map in
 // the 'frrState' struct.
-func sessionName(myAddr string, myAsn uint32, addr string, asn uint32) string {
-	return fmt.Sprintf("%d@%s-%d@%s", asn, addr, myAsn, myAddr)
+func sessionName(s session) string {
+	if s.VRFName == "" {
+		return fmt.Sprintf("%d@%s-%d@%s", s.PeerASN, s.PeerAddress, s.MyASN, s.SourceAddress)
```

Comment:
perhaps strings.builder can be used instead of having 2 similar fmt.sprintf?

Reply:
This function and the sprintf are compact enough that I wouldn't mind having a bit of duplication, however if that is a concern, I'd just go with string concatenation here.

I don't think this is so performance sensible that the benefit of using strings.Builder balances the slight penalty in readability, compared to just using +

Let me change it.

Reply:
done